src/som/primitives/TimerPrims.java

```diff
@@ -0,0 +1,171 @@
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import com.oracle.truffle.api.CompilerDirectives.CompilationFinal;
+import com.oracle.truffle.api.CompilerDirectives.TruffleBoundary;
+import com.oracle.truffle.api.RootCallTarget;
+import com.oracle.truffle.api.Truffle;
+import com.oracle.truffle.api.dsl.GenerateNodeFactory;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.source.SourceSection;
+
+import bd.primitives.Primitive;
+import som.compiler.AccessModifier;
+import som.interpreter.SomLanguage;
+import som.interpreter.actors.Actor;
+import som.interpreter.actors.EventualMessage.DirectMessage;
+import som.interpreter.actors.ReceivedMessage;
+import som.interpreter.actors.SFarReference;
+import som.interpreter.nodes.MessageSendNode;
+import som.interpreter.nodes.MessageSendNode.AbstractMessageSendNode;
+import som.interpreter.nodes.nary.BinaryComplexOperation.BinarySystemOperation;
+import som.interpreter.nodes.nary.UnaryExpressionNode.UnarySystemOperation;
+import som.vm.Symbols;
+import som.vm.VmSettings;
+import som.vm.constants.Classes;
+import som.vm.constants.Nil;
+import som.vmobjects.SBlock;
+import som.vmobjects.SInvokable;
+import som.vmobjects.SSymbol;
+import tools.concurrency.TraceParser;
+import tools.concurrency.TracingActors.ReplayActor;
+import tools.replay.actors.ActorExecutionTrace;
+import tools.replay.actors.ExternalEventualMessage.ExternalDirectMessage;
+import tools.replay.nodes.TraceActorContextNode;
+
+
+public final class TimerPrims {
+  @CompilationFinal private static Timer               timer;
+  @CompilationFinal private static ForkJoinPool        actorPool;
+  @CompilationFinal protected static Actor             timerActor;
+  @CompilationFinal protected static RootCallTarget    rct;
+  @CompilationFinal private static SomLanguage         language;
+  private static final SSymbol                         SELECTOR    =
+      Symbols.symbolFor("value");
+  private static final HashMap<Integer, SFarReference> targetMap   =
+      new HashMap<>();
+  @CompilationFinal private static List<Integer>       todolist;
+  private static AtomicInteger                         num         = new AtomicInteger(0);
+  private static final Object                          REPLAY_LOCK = new Object();
+
+  @GenerateNodeFactory
+  @Primitive(primitive = "actorSetupTimer:")
+  public abstract static class SetupTimerPrim extends UnarySystemOperation {
+    @Specialization
+    protected final Object setup(final Object o) {
+      timer = new Timer();
+      actorPool = vm.getActorPool();
+      language = vm.getLanguage();
+      timerActor = vm.getMainActor();
+      SInvokable s =
+          (SInvokable) Classes.blockClass.lookupMessage(SELECTOR, AccessModifier.PUBLIC);
+      rct = createOnReceiveCallTarget(SELECTOR,
```

Comment:
Any reason not to use the `ValueNonePrim` directly here?

Think, I'd just use that. Will perhaps also reduce code complexity here.

Reply:
Hm, I see, `AbstractMessageSendNode` is expected by `ReceivedMessage`.